src/main/java/app/bpartners/geojobs/endpoint/rest/security/BearerAuthFilter.java

```diff
@@ -0,0 +1,41 @@
+package app.bpartners.geojobs.endpoint.rest.security;
+
+import static app.bpartners.geojobs.endpoint.rest.security.authenticator.ApiKeyAuthenticator.API_KEY_HEADER;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.RequestMatcher;
+
+@Slf4j
+public class BearerAuthFilter extends AbstractAuthenticationProcessingFilter {
```

Comment:
Note that Bearer Authentication is different from ApiKey  Authentication. 
Bearer is generated randomly while Api Key is provided statically by the server to the client. 
Rename it as it must be.